File: CPP/PROGRAM/Komputer.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Cpu.cpp"
#include "Harddrive.cpp"
#include "Ram.cpp"
#include "Ssd.cpp"
#include "Keyboard.cpp"
#include "Mouse.cpp"
#include "Monitor.cpp"

using namespace std;

class Komputer{
    //atribut
    private:
        string nama;
        Cpu cpu;
        vector<Ram> ramList;
        Harddrive harddrive;
        Ssd ssd;
        Keyboard keyboard;
        Mouse mouse;
        Monitor monitor;

    public:
        //konstruktor
        Komputer(){
            this->nama = "";
        }

        Komputer(string nama, Cpu cpu, vector<Ram> ramList, Harddrive harddrive, Ssd ssd, Keyboard keyboard, Mouse mouse, Monitor monitor){
            this->nama = nama;
            this->cpu = cpu;
            this->ramList = ramList;
            this->harddrive = harddrive;
            this->ssd = ssd;
            this->keyboard = keyboard;
            this->mouse = mouse;
            this->monitor = monitor;
        }

        //getter dan setter
        void setNama(string nama){
            this->nama = nama;
        }

        void setCpu(Cpu cpu){
            this->cpu = cpu;
        }

        void setRam(vector<Ram> ramList){
            this->ramList = ramList;
        }    
    
        void setHarddrive(Harddrive harddrive){
            this->harddrive = harddrive;
        }

        void setSsd(Ssd ssd){
            this->ssd = ssd;
        }

        void setKeyboard(Keyboard keyboard){
            this->keyboard = keyboard;
        }

        void setMouse(Mouse mouse){
            this->mouse = mouse;
        }

        void setMonitor(Monitor monitor){
            this->monitor = monitor;
        }

        //add ram
        void addRam(Ram ram){
            this->ramList.push_back(ram);
        }

        string getNama(){
            return this->nama;
        }

        Cpu getCpu(){  
            return this->cpu;
        }

        vector<Ram> getRamList(){ // return vector
            return this->ramList;
        }

        Harddrive getHarddrive(){
            return this->harddrive;
        }

        Ssd getSsd(){  
            return this->ssd;
        }

        Keyboard getKeyboard(){  
            return this->keyboard;
        }

        Mouse getMouse(){  
            return this->mouse;
        }
        
        Monitor getMonitor(){  
            return this->monitor;
        }

        ~Komputer(){

        }
};

File: PYTHON/PROGRAM/Komputer.py
from Cpu import Cpu
from Harddrive import Harddrive
from Ram import Ram
from Ssd import Ssd
from Keyboard import Keyboard
from Mouse import Mouse
from Monitor import Monitor

class Komputer:
    def __init__(self, nama = "", cpu= None, ram_list = None,
                 harddrive = None, ssd = None, keyboard = None,
                 mouse = None, monitor = None):
        self.nama = nama
        self.cpu = cpu
        self.ram_list = ram_list if ram_list else []
        self.harddrive = harddrive
        self.ssd = ssd
        self.keyboard = keyboard
        self.mouse = mouse
        self.monitor = monitor

    # Setter
    def setNama(self, nama):
        self.nama = nama

    def setCpu(self, cpu):
        self.cpu = cpu

    def setRam_list(self, ram_list):
        self.ram_list = ram_list

    def setHarddrive(self, harddrive):
        self.harddrive = harddrive

    def setSsd(self, ssd):
        self.ssd = ssd

    def setKeyboard(self, keyboard):
        self.keyboard = keyboard

    def setMouse(self, mouse):
        self.mouse = mouse

    def setMonitor(self, monitor):
        self.monitor = monitor

    # add ram
    def addRam(self, ram):
        self.ram_list.append(ram)

    # Getter
    def getNama(self):
        return self.nama

    def getCpu(self):
        return self.cpu

    def getRam_list(self):
        return self.ram_list

    def getHarddrive(self):
        return self.harddrive

    def getSsd(self):
        return self.ssd

    def getKeyboard(self):
        return self.keyboard

    def getMouse(self):
        return self.mouse

    def getMonitor(self):
        return self.monitor
